src/algorithms/halley.jl

```diff
@@ -0,0 +1,12 @@
+"""
+    Halley(; concrete_jac = nothing, linsolve = nothing, linesearch = NoLineSearch(),
+        precs = DEFAULT_PRECS, autodiff = nothing)
+
+An experimental Halley's method implementation.
```

Comment:
Ig you will change this later. But keeping it marked so that it doesn't fall through.